src/dmd/parse.d

```diff
@@ -4447,7 +4447,33 @@ final class Parser(AST) : Lexer
                         parseAttributes;
 
                     AST.Declaration v;
-                    if (token.value == TOK.function_ ||
+                    AST.Dsymbol s;
+
+                    // allow declaring func type with the AliasDeclarationY flavor,
+                    // e.g `alias ident = Type (Parameters);`
+                    bool attributesAppended;
+                    const StorageClass funcStc = parseTypeCtor();
+                    Token* tlu = &token;
```

Comment:
No documentation on what the following grammar is being parsed.

Reply:
fixed